src/routes/donors.js

```diff
@@ -101,18 +101,54 @@ router.get('/sums', function(req, res, next) {
   })
 })
 
-router.put('/updatedonors/:id', (req, res) => {
-  console.log('hello world')
-  db.Donor.update(
-    {
-      email: req.body.email
-    },
-    {
-      where: {
-        id: req.params.id
-      }
-    }
-  ).then(result => res.json(result))
+router.get('/edit', (req, res) => {
+  db.PreferredContact.findAll({
+    attributes: ['id', 'description']
+  })
+    .then(result => {
+      res.json(result)
+    })
+    .catch(error => {
+      res.status(500).json(error)
+    })
+})
+
+router.put('/edit/:idNo', (req, res) => {
+  const { idNo: donorId } = req.params
+  const { remarks, preferredContact, dnc } = req.body
+  if (!remarks /* && !preferredContact && !('dnc' in req.body) */) {
+    return res.sendStatus(204).json({
+      message: 'You have not sent any parameters for updating'
+    })
+  }
+  try {
+    // We may or may not need to do this findOne, depending on whether the preferred contact is being updated.
+    let prom = Promise.resolve({})
+    prom
```

Comment:
Since you removed the `findOne`, this promise is always going to resolve with a value of `{}` (from the previous line). Therefore in line 128, `preferredContact` is going to be `undefined`. Your try block can just do:

```js
try {
  let updateRemarks = {}
  if (remarks) { ... }
  if (preferredContact) { // this is the preferredContact variable from line 118
    ...
  }
 ... 
  db.Donor
    .update(updateParams, { ... })
    .then(([, result]) => res.json(result))
} catch() { ... }
```